.github/workflows/workflow.yml

```diff
@@ -0,0 +1,81 @@
+name: Python API CI
+
+on: [push, pull_request]
+
+jobs:
+  build-from-source:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install From Source
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install ./PythonAPI
```

Comment:
Please follow the existing CI and add `--use-feature=in-tree-build`. Same below.

Reply:
FWIW, `in-tree-build` doesn't exist anymore. It was made the default in pip 21.3 and removed in 22.1.

Reply:
Ah, OK. Thanks for sharing that!